sale_order_line_cancel/models/sale_order_line.py

```diff
@@ -4,79 +4,27 @@
 # Copyright 2025 Michael Tietz (MT Software) <mtietz@mt-software.de>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import _, api, fields, models
-from odoo.tools import float_compare
+from odoo import models
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    product_qty_canceled = fields.Float(
-        "Qty canceled", readonly=True, copy=False, digits="Product Unit of Measure"
-    )
-    product_qty_remains_to_deliver = fields.Float(
-        string="Remains to deliver",
-        digits="Product Unit of Measure",
-        compute="_compute_product_qty_remains_to_deliver",
-        store=True,
-    )
-    can_cancel_remaining_qty = fields.Boolean(
-        compute="_compute_can_cancel_remaining_qty"
-    )
-
-    @api.depends("product_qty_remains_to_deliver", "state")
-    def _compute_can_cancel_remaining_qty(self):
-        precision = self.env["decimal.precision"].precision_get(
-            "Product Unit of Measure"
-        )
-        for rec in self:
-            rec.can_cancel_remaining_qty = (
-                float_compare(
-                    rec.product_qty_remains_to_deliver, 0, precision_digits=precision
-                )
-                == 1
-                and rec.state in ("sale", "done")
-                and rec.qty_delivered_method == "stock_move"
-            )
-
-    @api.depends("qty_to_deliver", "product_qty_canceled")
-    def _compute_product_qty_remains_to_deliver(self):
-        for line in self:
-            qty_remaining = max(0, line.qty_to_deliver - line.product_qty_canceled)
-            line.product_qty_remains_to_deliver = qty_remaining
-
     def _get_moves_to_cancel(self):
-        lines = self.filtered(lambda l: l.qty_delivered_method == "stock_move")
+        lines = self.filtered(
+            lambda l: l.qty_delivered_method == "stock_move"
+            and l.can_cancel_remaining_qty
+        )
         return lines.move_ids.filtered(lambda m: m.state not in ("done", "cancel"))
```

Comment:
On a performance point of view, this code iterates 2 times on the lines.... We should do it on 1 loop.

Reply:
Yes of course. I just copied the whole Module. And did Theo necessary Changes. 👍🏼

Reply:
Sorry no. It doesn't, or? :-)
At first it iterates on lines and then on the moves.